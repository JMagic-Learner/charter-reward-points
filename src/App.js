import logo from './logo.svg';
import './App.css';
import RewardsPage from './Component/RewardPage'

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <RewardsPage/>
      </header>
    </div>
  );
}

export default App;
