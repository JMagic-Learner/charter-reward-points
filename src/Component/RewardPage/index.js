import { toBeEmpty } from '@testing-library/jest-dom/dist/matchers'
import React, { useState, useEffect } from 'react'
import { mockData } from '../../API/mockData.api'


function RewardsPage() {

const [salesData,setSalesData] = useState([])
const [filterSales, setFilterSales] = useState([])
const [filterSales2, setFilterSales2] = useState([])
const [filterSales3, setFilterSales3] = useState([])
const [text, setText] = useState('');


useEffect(()=>{
mockData().then((data) => {
        console.log(data)
        let tempobject = data
        tempobject.forEach((item)=>{
            
            item.rewards = (item.price/100)*2 + (item.price/50)
            console.log(item)
        })
        console.log(tempobject)
        setSalesData(tempobject)
    });
},[])


const handleChange = event => {
    setText(event.target.value);
  };

function getMonthFromString(mon){
    return new Date(Date.parse(mon +" 1, 2022")).getMonth()+1
 }

function toMonthName(monthNumber) {
    const date = new Date();
    date.setMonth(monthNumber-1);
  
    return date.toLocaleString('en-US', {
      month: 'long',
    });
  }

function filterMonth(event) {
    event.preventDefault();

    // Lets define the months to be displayed // 
    let monthNumber = getMonthFromString(text)
    let monthHistory = [monthNumber, monthNumber-1, monthNumber-2]
    let monthLimit = monthHistory.map((item)=>{
        return(
        toMonthName(item)
        )
    })

    for (let i = 0; i<= monthLimit.length-1; i++ ) {
        let filtered = salesData.filter((item=>item.month=== monthLimit[i]))
        if (i===0) {
            
            setFilterSales(filtered)
        } else if ( i ===1) {
            setFilterSales2(filtered)
        } else {
        setFilterSales3(filtered)
        }
    }
        
    
    
    
}

function show(event) {
    event.preventDefault();
    setFilterSales(salesData)
    setFilterSales2([])
    setFilterSales3([])
}


    return(
        <div className="rewards-homepage">
            <p> Rewards </p>
                
                <input
                    type="text"
                    id="message"
                    name="filterName"
                    onChange={handleChange}
                    placeholder="Search by Month"
                    value={text}
                />
                <button type="submit" onClick={filterMonth}> Submit </button>
                <button type="submit" onClick={show}> Show All </button>
                <div className="display-all-customers"> 
                <table>
                <tr className="rewards-row">
                        <th className="rewards-cell">Month</th>
                        <th className="rewards-cell">Customer </th>
                        <th className="rewards-cell">Spent </th>
                        <th className="rewards-cell">Rewards </th>
                </tr>
                    {filterSales.map((item) => {
                        return( 
                        <tr> 
                            <td className="rewards-cell">{item.month}</td>
                            <td className="rewards-cell">{item.name}</td>
                            <td className="rewards-cell">{item.price}</td>
                            <td className="rewards-cell">{item.rewards}</td>
                        </tr>
                        )
                    })}
                     {filterSales2.map((item) => {
                        return( 
                        <tr> 
                            <td className="rewards-cell">{item.month}</td>
                            <td className="rewards-cell">{item.name}</td>
                            <td className="rewards-cell">{item.price}</td>
                            <td className="rewards-cell">{item.rewards}</td>
                        </tr>
                        )
                    })}
                    {filterSales3.map((item) => {
                        return( 
                        <tr> 
                            <td className="rewards-cell">{item.month}</td>
                            <td className="rewards-cell">{item.name}</td>
                            <td className="rewards-cell">{item.price}</td>
                            <td className="rewards-cell">{item.rewards}</td>
                        </tr>
                        )
                    })}
                    </table>
                </div>
        </div>
    )
}

export default RewardsPage
