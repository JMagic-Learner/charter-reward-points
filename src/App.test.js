import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import RewardPage from "./Pages/RewardPage/index.js"
import App from './App';

describe ( `RewardPage`, ()=> {
  it('should match snapshot', () => {
    })
    const { asFragment } = render(<RewardPage > SnapShot Rewardpage </RewardPage>);
    expect(asFragment()).toMatchSnapshot();
})
