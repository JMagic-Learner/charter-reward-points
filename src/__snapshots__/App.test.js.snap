// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<DocumentFragment>
  <div
    class="rewards-homepage"
  >
    <p>
       Rewards 
    </p>
    <input
      id="message"
      name="filterName"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
       Submit 
    </button>
    <button
      type="submit"
    >
       Show All 
    </button>
    <div
      class="display-all-customers"
    >
      <table>
        <tbody>
          <tr
            class="rewards-row"
          >
            <th
              class="rewards-cell"
            >
              Month
            </th>
            <th
              class="rewards-cell"
            >
              Customer 
            </th>
            <th
              class="rewards-cell"
            >
              Spent 
            </th>
            <th
              class="rewards-cell"
            >
              Rewards 
            </th>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
